purchase_request_procurement/models/procurement.py

```diff
@@ -84,12 +84,13 @@ def propagate_cancels(self):
             # Remove the purchase request lines, if the request is not draft
             # or reject
             for line in request_lines:
-                if line.request_id.state not in ('draft', 'reject'):
-                    raise ValidationError(_('Can not cancel this procurement '
-                                            'as the related purchase request '
-                                            'is in progress confirmed already.'
-                                            ' Please cancel the purchase'
-                                            ' request first.'))
+                if (line.request_id.state not in ('draft', 'reject') and not
+                        self._context.get('from_purchase_request')):
```

Comment:
You are right about the typo error, It something that was in the original module, I take the ocassion to fix. Good catch!

In the other hand, `from_purchase_request` context has to be checked because this is what makes this error raise if you try to directly cancel the procurement and allows you to cancel when rejecting a purchase request.